src/lib/relative_attr.js

```diff
@@ -0,0 +1,47 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+
+var ASCEND = /^(.*)(\.[^\.\[\]]+|\[\d\])$/;
```

Comment:
Could you add a very short comment about what this matches? In particular figuring out whether `^` is negation or an actual carat takes a good amount of staring and recalling js-specific regex escaping rules since this particular use could very reasonably involve either one.

Reply:
(re-reading, matches `foo.string` or `foo[5]` or `foo.bar[5]`, I think, while capturing the final part as well as everything before it. It's not so bad, I guess.